test/subset-band-name/subset_band_name_utitlities.py

```diff
@@ -35,7 +59,32 @@ def remove_results_files() -> None:
 
 def compare_data(reference_file: str, test_file: str, sds_name: str) -> bool:
     """Compares two data dimension sizes"""
-    reference_data = get_sds_data(reference_file, sds_name)
-    test_data = get_sds_data(test_file, sds_name)
 
-    return numpy.array_equal(reference_data, test_data)
+    # Compare one variable at a time
+    if sds_name is not None:
+        reference_data = get_sds_data(reference_file, sds_name)
+        test_data = get_sds_data(test_file, sds_name)
+
+        return numpy.array_equal(reference_data, test_data)
+
+    # Compare all variables and select VData in files
+    else:
+        reference_sds_names = get_all_sds_names(reference_file)
+
+        for sds_name in reference_sds_names:
```

Comment:
`get_all_sds_names` seems to be returning a dictionary, and this works, but it might be clearer if you were iterating over the keys()
```suggestion
        for sds_name in reference_sds_names.keys():
```

On further thinking you should probably change the output of `get_all_sds_names` so it's really just returning the names.

Reply:
Nice catch. Thanks!